simplekv/decorator.py

```diff
@@ -100,3 +101,39 @@ def _unmap_key(self, key):
         assert key.startswith(self.prefix)
 
         return key[len(self.prefix):]
+
+
+class URLEncodeKeysDecorator(KeyTransformingDecorator):
+    """URL-encodes keys before passing them on to the underlying store."""
+    def _map_key(self, key):
+        if not isinstance(key, text_type):
+            raise ValueError('%r is not a unicode string' % key)
+        quoted = quote_plus(key.encode('utf-8'))
+        if isinstance(quoted, binary_type):
+            quoted = quoted.decode('utf-8')
+        return quoted
+
+    def _unmap_key(self, key):
+        return unquote_plus(key)
+
+
+class ReadOnlyStoreDecorator(StoreDecorator):
```

Comment:
Maybe this is a good time to change the name to ReadOnlyDecorator, to be more consistent with the other decorators